src/source_util.cpp

```diff
@@ -74,68 +74,32 @@ bool sourceRangeContainsOnly(clang::SourceLocation beginLocation,
   return true;
 }
 
-// Find the first r-paren token that balances the l-paren at the beginning of
-// the list. If no such token exists, return a past the end iterator
-llvm::ArrayRef<clang::Token>::const_iterator
-skipToBalancedRParenTok(llvm::ArrayRef<clang::Token>::const_iterator begin,
-                        llvm::ArrayRef<clang::Token>::const_iterator end,
-                        const clang::SourceManager &SM) {
-  if (begin == end) {
-    return end;
-  }
-
-  assert(begin->is(clang::tok::l_paren) &&
-         "First iterator must point to an lparen token! ");
-
-  int leftBalance = 0;
-  clang::LangOptions langOpts;
-
-  // If there is an l_paren token after 'start':
-  // find the r_paren that balances it within that
-  // subexpression.
-  for (auto it = begin; it != end; ++it) {
-    if (it->is(clang::tok::l_paren)) {
-      ++leftBalance;
-    } else if (it->is(clang::tok::r_paren)) {
-      --leftBalance;
-    }
-
-    // If balanced, return the iterator pointing to the r-paren
-    if (leftBalance == 0) {
-      return it;
-    }
-  }
-
-  // Default to returning the end iterator in case we never hit 0 balance in
-  // the preceeding loop
-  return end;
-}
-
 } // namespace
 
 clang::CharSourceRange
 SourceUtil::expandRange(const clang::SourceRange &range,
                         const clang::SourceManager &sourceManager) {
   // Get the start location, resolving from macro definition to macro call
-  // location. Special handling is needed for a statement in a macro body, we
-  // want to resolve to the entire macro call.
+  // location. The loop is adapted from 'clang::SourceManager::getFileLoc'.
 
   clang::SourceLocation begin = range.getBegin();
-  if (sourceManager.isMacroBodyExpansion(begin)) {
-    begin = sourceManager.getExpansionRange(begin).getBegin();
-  } else {
-    begin = sourceManager.getFileLoc(begin);
+  while (!begin.isFileID()) {
+    if (sourceManager.isMacroArgExpansion(begin))
```

Comment:
Style nit, lets keep the braces after the `if/else` parts.

Reply:
Fixed